Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -397,6 +403,10 @@ public class ResponseAccumulator: HTTPClientResponseDelegate {
         case .head(let head):
             self.state = .body(head, part)
         case .body(let head, var body):
+            if body.writerIndex + part.readableBytes > UInt32.max {
```

Comment:
`UInt32.max` is a bit magic. I think the best solution would be to add a static property in `swift-nio` to `ByteBuffer` with the max size it can handle. But we can also just give this constant a descriptive name here and document it.